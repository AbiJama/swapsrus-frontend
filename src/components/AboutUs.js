import React from "react";
import { Link } from "react-router-dom";
import "../styles/about-us.css";

function AboutUs({ Toys, addToys })  {
  return (
    <div className="about-us">
      <h1>About Us</h1>
      <h2 className="about-us_tag">Borrow. Play. Repeat.</h2>
      <div className="about-us_text">
      <p>We are a toy swapping website for local communities.</p>
      <p>Toys can be expensive and children become easily bored of playing with them leaving you with rooms full of pretty new unloved toys.</p> 
      <p>At Swaps R Us you can upload your lonely toys and let others borrow them and you can pick some to borrow too.</p>
      <p>Contact the lender and arrange a local pick-up. Swap and enjoy and then when you are bored - swap again!</p>
      <p>Browse and upload your toys below.</p>
      </div>
      <div className="button-container">
        <Link to="/toys">
          <button type="button" className="browse-toys-logo" onClick={Toys}>
            <img className="about-us_button" src="browse-toys.png" alt="logo with browse toys text" />
          </button>
        </Link>
        <Link to="/add-toys">
          <button type="button" className="upload-toys-logo" onClick={addToys}>
            <img className="about-us_button" src="upload-toys.png" alt="logo with upload toys text" />
          </button>
        </Link>
      </div>
    </div>
  );
};

export default AboutUs;

