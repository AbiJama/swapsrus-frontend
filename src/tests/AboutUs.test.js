import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AboutUs from "../components/AboutUs";

describe("AboutUs component", () => {
  it("should navigate to /toys when browse toys button is clicked", () => {
    const { getByAltText } = render(
      <MemoryRouter>
        <AboutUs />
      </MemoryRouter>
    );

    const browseToysButton = getByAltText("logo with browse toys text");

    fireEvent.click(browseToysButton);

    expect(window.location.pathname).toBe("/toys");
  });

  it("should navigate to /add-toys when upload toys button is clicked", () => {
    const { getByAltText } = render(
      <MemoryRouter>
        <AboutUs />
      </MemoryRouter>
    );

    const uploadToysButton = getByAltText("logo with upload toys text");

    fireEvent.click(uploadToysButton);

    expect(window.location.pathname).toBe("/add-toys");
  });
});
