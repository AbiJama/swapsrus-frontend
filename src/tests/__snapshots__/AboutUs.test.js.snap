// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs renders correctly 1`] = `
<DocumentFragment>
  <div
    class="about-us"
  >
    <h1>
      About Us
    </h1>
    <h2
      class="about-us_tag"
    >
      Borrow. Play. Repeat.
    </h2>
    <div
      class="about-us_text"
    >
      <p>
        We are a toy swapping website for local communities.
      </p>
      <p>
        Toys can be expensive and children become easily bored of playing with them leaving you with rooms full of pretty new unloved toys.
      </p>
      <p>
        At Swaps R Us you can upload your lonely toys and let others borrow them and you can pick some to borrow too.
      </p>
      <p>
        Contact the lender and arrange a local pick-up. Swap and enjoy and then when you are bored - swap again!
      </p>
      <p>
        Browse and upload your toys below.
      </p>
    </div>
    <div
      class="button-container"
    >
      <a
        href="/toys"
      >
        <button
          class="browse-toys-logo"
          type="button"
        >
          <img
            alt="logo with browse toys text"
            class="about-us_button"
            src="browse-toys.png"
          />
        </button>
      </a>
      <a
        href="/add-toys"
      >
        <button
          class="upload-toys-logo"
          type="button"
        >
          <img
            alt="logo with upload toys text"
            class="about-us_button"
            src="upload-toys.png"
          />
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
